File: harvey/10828.py
n = int(input())

stack = []

for i in range(n):
    command, *arg = input().split()
    if command=='push':
        stack.append(arg[0])
    elif command=='pop':
        print(stack.pop() if stack else -1)
    elif command=='size':
        print(len(stack))
    elif command=='empty':
        print(0 if stack else 1)
    elif command=='top':
        print(stack[-1] if stack else -1)

File: chu/10828.java
package week2;

import java.util.*;
public class aa10828 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		Stack st = new Stack();
		
		int tc = sc.nextInt();
		
		for(int i = 0; i <= tc; ++i){
			String stdin = sc.nextLine();
			if(stdin.length() > 0){
				int val = 0;
				String stack = stdin.split(" ")[0];
				if(stdin.length() > 5){
					val = Integer.parseInt(stdin.split(" ")[1]);
				}
					switch(stack){
				case "push":
					st.push(val);
					break;
				case "top":
					if(st.isEmpty())
						System.out.println(-1);
					else
						System.out.println(st.peek());
					break;
				case "size":
					System.out.println(st.size());
					break;
				case "empty":
					if(st.isEmpty())
						System.out.println(1);
					else
						System.out.println(0);
					break;
				case "pop":
					if(st.isEmpty())
						System.out.println(-1);
					else
						System.out.println(st.pop());
					break;
				}
			}	
		}
	}
}
